### PR TITLE
[MI-2615]: Fixed the placeholder of modal components like Autosuggest, Input field, and Textarea

### PR DESCRIPTION
**Pull request by Nityanand13**
Screenshot from 2023-02-16 14-33-06
Screenshot from 2023-02-16 14-33-34

### CODE REVIEW

File: src/components/AutoSuggest/index.tsx [hunk 1]
```diff
@@ -156,14 +156,18 @@ const AutoSuggest = ({
                     onClick={() => setFocused(true)}
                     readOnly={true}
                 />
+                {inputValue?.length>0 && <label className={'margin-0 wt-400 auto-suggest__field-placeholder'}>
+                    {placeholder}
+                    {required && '*'}
+                </label>}
```
Comment (ayusht2810) [inline src/components/AutoSuggest/index.tsx:162]:
```suggestion
                {inputValue?.length>0 && (
                    <label className={'margin-0 wt-400 auto-suggest__field-placeholder'}>
                        {placeholder}
                        {required && '*'}
                    </label>
                    )}
```

File: src/components/AutoSuggest/index.tsx [hunk 2]
```diff
@@ -156,14 +156,18 @@ const AutoSuggest = ({
                     onClick={() => setFocused(true)}
                     readOnly={true}
                 />
+                {inputValue?.length>0 && <label className={'margin-0 wt-400 auto-suggest__field-placeholder'}>
```
Comment (ayusht2810) [inline src/components/AutoSuggest/index.tsx:159]:
```suggestion
                {inputValue?.length && <label className={'margin-0 wt-400 auto-suggest__field-placeholder'}>
```

Comment (ayusht2810) [reply src/components/AutoSuggest/index.tsx:159]: check if this works

File: src/components/AutoSuggest/index.tsx [hunk 3]
```diff
@@ -156,14 +156,18 @@ const AutoSuggest = ({
...
+                </label>}
                 <input
                     ref={textInputFieldRef}
                     placeholder={`${placeholder ?? ''}${required ? '*' : ''}`}
                     value={inputValue}
                     onChange={(e) => onInputValueChange(e.target.value)}
                     onKeyDown={keyboardNavigation}
                     onBlur={handleBlur}
-                    className={`auto-suggest__input padding-0 ${disabled && 'cursor-not-allowed'}`}
+                    className={`auto-suggest__input ${disabled && 'cursor-not-allowed'} ${inputValue?.length>0 && 'auto-suggest__input--shifted'}`}
```
Comment (ayusht2810) [inline src/components/AutoSuggest/index.tsx:170]: same

File: src/components/AutoSuggest/index.tsx [hunk 4]
```diff
@@ -193,7 +197,11 @@ const AutoSuggest = ({
             </ul>
             {selectedOption && (
                 <div className='d-flex align-items-center justify-content-between auto-suggest__selected-option-container padding-h-20 channel-bg'>
-                    <div className='auto-suggest__selected-option-value text-ellipsis'>{renderValue(selectedOption)}</div>
+                    {inputValue?.length>0 && <label className={'margin-0 wt-400 auto-suggest__field-placeholder'}>
+                        {placeholder}
+                        {required && '*'}
+                    </label>}
```
Comment (ayusht2810) [inline src/components/AutoSuggest/index.tsx:203]: same. fix everywhere

File: src/components/AutoSuggest/index.tsx [hunk 5]
```diff
@@ -193,7 +197,11 @@ const AutoSuggest = ({
...
+                    <div className={`auto-suggest__selected-option-value text-ellipsis ${inputValue?.length>0 && 'auto-suggest__input--shifted'}`}>{renderValue(selectedOption)}</div>
```
Comment (ayusht2810) [inline src/components/AutoSuggest/index.tsx:204]:
```suggestion
                    <div className={`auto-suggest__selected-option-value text-ellipsis ${inputValue?.length && 'auto-suggest__input--shifted'}`}>
                       {renderValue(selectedOption)}
                    </div>
```

File: src/components/InputField/index.tsx [hunk 6]
```diff
@@ -26,13 +26,17 @@ const Input = ({
 }: InputFieldProps) => (
     <div className={`form-group ${className}`}>
         {label && <label className='form-group__label wt-400'>{label}</label>}
+        {value.length>0 && <label className={'wt-400 form-group__field-placeholder'}>
```
Comment (ayusht2810) [inline src/components/InputField/index.tsx:29]:
```suggestion
        {value.length>0 && <label className='wt-400 form-group__field-placeholder'>
```

Comment (ayusht2810) [reply src/components/InputField/index.tsx:29]: fix everywhere

File: src/styles/components/_autoSuggest.scss [hunk 7]
```diff
@@ -36,6 +36,17 @@
       background: transparent;
       outline: none;
     }
+
+    &--shifted {
+      padding-top: 10px;
+    }
+  }
+  &__field-placeholder {
```
Comment (ayusht2810) [inline src/styles/components/_autoSuggest.scss:44]:
```suggestion
  }
  
  &__field-placeholder {
```

**Review by ayusht2810 — APPROVED** [review]

**Review by ayusht2810 — DISMISSED** [review]
Update the version of the library.

File: src/components/AutoSuggest/index.tsx [hunk 8]
```diff
@@ -156,14 +156,20 @@ const AutoSuggest = ({
                     onClick={() => setFocused(true)}
                     readOnly={true}
                 />
+                {!!inputValue?.length && (
```
Comment (sooraj-shukla) [inline src/components/AutoSuggest/index.tsx:159]: Please update the PR description and add screenshots if the UI is updated.

File: src/styles/components/_autoSuggest.scss [hunk 9]
```diff
@@ -36,6 +36,18 @@
...
+
+    &--shifted {
+      padding-top: 10px;
+    }
+  }
+
+  &__field-placeholder {
+    max-width: calc(100% - 20px);
+    color: rgba(var(--center-channel-color-rgb), 0.4);
+    @include positioning(absolute, 10px, '', '', 17px, 0);
```
Comment (sooraj-shukla) [inline src/styles/components/_autoSuggest.scss:48]: Include mixins to top.

File: src/styles/components/_textArea.scss [hunk 10]
```diff
@@ -31,4 +33,21 @@
...
+    max-width: calc(100% - 20px);
+    color: rgba(var(--center-channel-color-rgb), 0.4);
+    @include positioning(absolute, 8px, '', '', 11px, 0);
+    transition: all 200ms ease-in-out;
+    font-size: 90%;
+    padding-top: 6px;
+    width: 100%;
+    display: block;
+    background: white;
+    top: 6px;
```
Comment (sooraj-shukla) [inline src/styles/components/_textArea.scss:47]: Can't we give the top using `positioning` mixin.

File: src/styles/components/_textArea.scss [hunk 11]
```diff
@@ -31,4 +33,21 @@
...
+
+  &__field-placeholder {
+    max-width: calc(100% - 20px);
+    color: rgba(var(--center-channel-color-rgb), 0.4);
+    @include positioning(absolute, 8px, '', '', 11px, 0);
+    transition: all 200ms ease-in-out;
+    font-size: 90%;
+    padding-top: 6px;
+    width: 100%;
+    display: block;
```
Comment (sooraj-shukla) [inline src/styles/components/_textArea.scss:45]: Can't we use a util class for this?

**Review by sooraj-shukla — CHANGES_REQUESTED** [review]

**Review by sooraj-shukla — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**